Adding overwrite_config option to build_env_pointnet() and merging 2024 bmtk-workshop opt tutorials

beaherrera opened the pull request:
- Adding overwrite_config option to build_env_pointnet(). It returned an error when rebuilding the environment with different config options for the same base_dir. This option was already available for build_env_bionet().
- Merging optional tutorials from 2024 bmtk-workshop into [Additional Tutorials](https://alleninstitute.github.io/bmtk/ad_tutorials.html) on the bmtk documentation. I also modified the autodocs scripts and pages to update the website. 
- Updated [Annual Modeling Software Workshop](https://alleninstitute.github.io/bmtk/workshop.html) page: 
    - Added the link to the 2024 workshop recordings and the signed-up link for the newsletter.

beaherrera committed:
Adding overwrite_config option to build_env_pointnet()

beaherrera committed:
Updated workshop page on documentation

Added link to 2024 workshop videos and newsletter signup

beaherrera committed:
Merging opt tutorials from 2024 bmtk-workshop

- Updated format to match main bmtk tutorials
- Fixed typos

beaherrera committed:
Adding opt tutorials to 'Additional Tutorials' in doc

- added opt tutorials to ad_tutorials.rst
- fixed bugs for getting opt tutorial notebooks to work with Sphinx

beaherrera committed:
Fixed bug and typos in Ch_advanced_spikes_input notebook

kaeldai commented:
Hi @beaherrera, I noticed that in the changes you made to the sim_setup.py and conf.py all seem to have changed single quotes (') to double quotes (").

Python doesn't make the distinction between the two, and I personally don't have any hard-and-fast rule on when and how which quotes to use (although I tend to prefer single quotes myself). Is it possible this was something changed by your IDE? If so you will want to disable that because, besides making it hard to look at changes, if you aren't careful automatically changing quote styles can cause other issues.

beaherrera commented:
Hi @kaeldai,

Thank you for reviewing the PR! I'm sorry about that. It’s the auto formatter I use. I’ll fix the commit. I just changed a few lines, which are hard to see among all the changes. 

Best,
Beatriz

beaherrera committed:
Reverting commit d6f055941264ef4327212284983c51b441f3683d

Removing Black autoformatting

beaherrera committed:
Adding overwrite_config option to build_env_pointnet()

beaherrera commented:
Hi @kaeldai,

I reverted the commit and removed the auto string normalization. You can see the changes I made to build_env_pointnet() in my last commit [1698d2c](https://github.com/AllenInstitute/bmtk/pull/386/commits/1698d2ce685d7d4700e8c9231e9de41270fedf7e)

Thanks,
Beatriz

beaherrera committed:
Update Ch_External_Models.ipynb

beaherrera committed:
Revert "Updated workshop page on documentation"

This reverts commit cdb42ced148724a9231cda77f875cb86ca7f9f5b.

beaherrera committed:
Resolving conflicts in PR#386

beaherrera commented:
Hi @kaeldai,

I reverted the commits to update the old documentation website since they are no longer needed and were causing conflicts.

Updated summary of changes:
- Adding overwrite_config option to build_env_pointnet(). It returned an error when rebuilding the environment with different config options for the same base_dir. This option was already available for build_env_bionet().
- Adding optional tutorials from 2024 bmtk-workshop to [docs/tutorial](https://github.com/AllenInstitute/bmtk/tree/develop/docs/tutorial) folder. I haven't added these tutorials to the documentation itself yet.

Thanks,
Beatriz

kaeldai commented:
Thanks, if you are finished with these changes feel free to merge the PR!

beaherrera commented:
Thanks, @kaeldai!
Ok, sounds good!
